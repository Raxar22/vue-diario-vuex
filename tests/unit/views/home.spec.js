import { shallowMount } from "@vue/test-utils"
import Home from '@/views/HomeView'

describe ('Pruebas en el Home Vie', () => {

    test('Debe renderizar el componente correctamente',()=>{
        const wrapper = shallowMount(Home)
        expect(wrapper.html()).toMatchSnapshot()
    })

    test('Hacer click debe redireccionar a NoEntradas', () =>{
        const mockRouter = {
            push: jest.fn()
        }
        const wrapper = shallowMount(Home, {
            global: {
                mocks:{
                    $router: mockRouter
                }
            }
        })

        wrapper.find('button').trigger('click')
        expect(mockRouter.push).toHaveBeenCalled()
        expect(mockRouter.push).toHaveBeenCalledWith({name:'NoEntradas'})

    })
})