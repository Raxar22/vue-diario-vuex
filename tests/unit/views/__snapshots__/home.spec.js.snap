// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Home Vie Debe renderizar el componente correctamente 1`] = `
<div class="home"><img alt="Vue logo" src="">
  <h1> Hola mundo </h1><button class="btn btn-primary"> Primero </button><button class="btn btn-secondary"> Segundo </button><button class="btn btn-success"> Exito </button>
</div>
`;
